### PR TITLE
[wip] Add support for choo nodes

### PR DESCRIPTION
**Pull request by yoshuawuyts**
Work in progress patch for adding [`choo`](https://github.com/yoshuawuyts/choo) support. This would shave off ~`3kb` of the resulting bundle (`7.2kb` -> `4.6kb`) making it one of the tiniest full-featured frameworks out there.

I hope this patch is acceptable; I think the added overhead to support `choo` too would be super little, and would help me out heaps!

I'm struggling a bit with getting this to work; if you spot any obvious mistakes: all help is welcome. I'll hopefully have this running super soon. Thanks!
- related issue on choo https://github.com/yoshuawuyts/choo/issues/3

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -49,8 +49,16 @@ module.exports = function (file, opts) {
     this.push(null)
   }
   function walk (node) {
-    if (node.type === 'TemplateLiteral' &&
-      node.parent.tag && isBelOrYoYo.indexOf(node.parent.tag.name) !== -1) {
+    if (node.type === 'TemplateLiteral' && node.parent.tag &&
+      node.parent.tag.object &&
+      isBelOrYoYo.indexOf(node.parent.tag.object.name) !== -1) {
+      const index = isBelOrYoYo.indexOf(node.parent.tag.object.name)
+      const name = isBelOrYoYo[index]
```
Comment (yoshuawuyts) [inline index.js:56]: could probably find the parent node's name in less characters; might lead to cleaner code

File: package.json [hunk 2]
```diff
@@ -34,7 +34,9 @@
   },
   "devDependencies": {
     "browserify": "^13.0.0",
+    "choo": "^1.1.0",
     "standard": "^6.0.8",
-    "tape": "^4.5.1"
+    "tape": "^4.5.1",
+    "yo-yo": "^1.2.0"
```
Comment (yoshuawuyts) [inline package.json:40]: should probably also add a regression test for `yo-yo`; perhaps in a separate patch

**Comment by shama** [comment]
👍 to this, thanks!

**Comment by shama** [comment]
Heads up, this library is turning into a plugin that has the transform applied globally. This way we can further optimize the size of the outputted build. Hopefully it will make it easier to implement this as well.

**Comment by yoshuawuyts** [comment]
@shama ahhh, rad! I'll give it another go then :sparkles:

**Comment by yoshuawuyts** [comment]
Closing as https://github.com/yoshuawuyts/choo/issues/71 will make it easier and https://github.com/shama/yo-yoify/issues/13 is a bit of blocker rn. Creating an issue instead for now

**Pull request closed without merging**